### PR TITLE
Searchable tokenization implementation

### PR DESCRIPTION
**Pull request by Zhaars**
Current tokenization simply replaces data when we write or read. The idea of this PR is to make it possible to replace the data even in the search requests, thus the search will work during tokenization, not only encryption. We need to replace type `column1=1234` with `column1=54872`, where 348729 is the corresponding tokenized value if `consistent_tokenization` setting is provided.

This PR contains an implementation of searchable tokenization for PostgreSQL for text/binary protocols.

## Checklist

- [X] Change is covered by automated tests
- [X] The [coding guidelines] are followed
- [ ] Public API has proper documentation in the [Acra documentation] site or has PR on [documentation repository] 
  with new changes
- [ ] CHANGELOG.md is updated (in case of notable or breaking changes)
- [ ] CHANGELOG_DEV.md is updated
- [ ] Benchmark results are attached (if applicable)
- [ ] [Example projects and code samples] are up-to-date (in case of API changes) 

[coding guidelines]: https://golang.org/doc/effective_go
[Example projects and code samples]: https://github.com/cossacklabs/acra-engineering-demo
[Acra documentation]: https://docs.cossacklabs.com/
[documentation repository]: https://github.com/cossacklabs/product-docs

### CODE REVIEW

File: pseudonymization/tokenizeQuery_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	randomRead(randomBytes)
+
+	type testcase struct {
+		Value     []byte
+		Type      common.TokenType
+		TokenType string
+		Query     string
+	}
+	testcases := []testcase{
+		{Value: []byte("somedata"), Type: common.TokenType_String, TokenType: "str", Query: "select data1 from test_table where data1='somedata'"},
```
Comment (Lagovas) [inline pseudonymization/tokenizeQuery_test.go:76]: lets test a bit more complicated queries with several searchable fields including several that should be ignored? to be sure, that we handle not only the simplest case but in general too.

File: pseudonymization/tokenizeQuery.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+			logrus.WithError(err).WithField("index", valueIndex).Debug("Failed to encrypt column")
+			return values, false, err
+		}
+		// it is ok to ignore the error if not column setting provided
+		_ = newValues[valueIndex].SetData(tokenized, encryptionSetting)
+	}
+	return newValues, true, nil
+}
+
+func (encryptor *TokenizeQuery) tokenizerDataWithSetting(setting config.ColumnEncryptionSetting) func(ctx context.Context, dataToTokenize []byte) (tokenized []byte, err error) {
```
[the remark this answers is not in the document]
Comment (Zhaars) [reply pseudonymization/tokenizeQuery.go:177]: Yes, the idea to return a function that can closure `config.ColumnEncryptionSetting`

File: pseudonymization/tokenizeQuery_test.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		schema, err := config.MapTableSchemaStoreFromConfig([]byte(fmt.Sprintf(schemaConfigTemplate, tcase.TokenType)))
+		assert.NoError(t, err)
+
+		encryptor := NewPostgresqlTokenizeQuery(schema, tokenEncryptor)
+
+		setting := config.BasicColumnEncryptionSetting{
+			TokenType:              tcase.TokenType,
+			ConsistentTokenization: true,
+		}
+		anonimized, err := tokenizer.Tokenize(tcase.Value, common.TokenContext{ClientID: []byte("client-id")}, &setting)
```
[the remark this answers is not in the document]
Comment (Zhaars) [reply pseudonymization/tokenizeQuery_test.go:93]: Makes sense!

**Review by Lagovas — APPROVED** [review]
lgtm

**Pull request merged**